SparkPost/2web2ui
# FE-1266 - Add aggregate metrics section

FE-1266 - Add aggregate metrics section

### What Changed

- Used AggregateMetrics and CompareByAggregateMetrics  for displaying metrics on dashboard
- Removed useReportBuilderContext from AggregateMetrics and CompareByAggregateMetrics and used props instead
- updated usePinnedReport to get the correct reportOption when report is changed.

### How To Test
- Verify with the report builder page that metrics for all the reports look correct on dashboard too.
- Verify with the report builder page that chart for all the reports look correct on dashboard too.
- Add the metrics section as in the mocks - https://sparkpost.invisionapp.com/console/share/RU1SP41BG7/483505018
- if no report is pinned (user ui options "pinned_report"), these metrics are default metrics for Summary Report
- if user ui options has a "pinned_report", then these metrics are the one associated with pinned report.

### To Do

- [x] Move AggregateMetrics and CompareByAggregateMetrics into src/components/reportBuilder
